### DISCUSSION TITLE
My RLS is not working though.

### DISCUSSION THREAD

**Discussion by myagizmaktav** [supabase/supabase#11408] [Questions] [answered]
![image](https://user-images.githubusercontent.com/115627863/210260003-59c9032b-e1b4-44ed-88a6-5340a4f5ff9f.png)

![image](https://user-images.githubusercontent.com/115627863/210260065-ef1b7054-0b12-4412-bb7c-02c2e5dc734b.png)

![image](https://user-images.githubusercontent.com/115627863/210260106-58649982-93fc-43bb-b450-98ea3b198b39.png)

my userdata
```json
{
  id: '363874c2-acf9-45f8-acfc-ef637965346b',
  aud: 'authenticated',        
  role: 'authenticated',       
  email: 'mym@admin.com',      
  email_confirmed_at: '2022-12-24T12:59:02.934117Z',
  phone: '',
  confirmed_at: '2022-12-24T12:59:02.934117Z',
  last_sign_in_at: '2023-01-02T16:47:00.348319745Z',
  app_metadata: {
    claims_admin: true,        
    provider: 'email',
    providers: [ 'email' ],    
    userlevel: 200,
    userrole: 'admin'
  },
  user_metadata: { age: 27, first_name: 'John' },
  identities: [
    {
      id: '363874c2-acf9-45f8-acfc-ef637965346b',
      user_id: '363874c2-acf9-45f8-acfc-ef637965346b',        
      identity_data: [Object], 
      provider: 'email',       
      last_sign_in_at: '2022-12-24T12:59:02.932703Z',
      created_at: '2022-12-24T12:59:02.932739Z',
      updated_at: '2022-12-24T12:59:02.932739Z'
    }
  ],
  created_at: '2022-12-24T12:59:02.929607Z',
  updated_at: '2023-01-02T16:47:00.350338Z'
}
```
my discord help chat discuss id  1056508114251300864
First I try with uid throughtout photo 2 is not working then I discuss dicord. Gary say to me  claims
https://github.com/supabase-community/supabase-custom-claims 
and I try and its not working to (I opened issue for this reason).
At last I try jwt style because I see in forum somebody uses like that. It's not working to.
I was stuck for 2 weeks how did I solve this solution I really don't know.
Thanks for the helping.

**Comment by GaryAustin1** [comment, marked as answer]
This was solved in Discord with setting persistSession:false as the OP was using vscode which (I assume) does not have local storage.
If anyone knows vscode typescript environment and a way to add local support this user would probably benefit.

Also turns out the OP was on 2.2.1 supabase-js which had this issue:
https://github.com/supabase/gotrue-js/issues/539
It is claimed to be "fixed" in 2.2.3, but the OP did not confirm with 2.2.3.
